src/components/ui/submit-button/submit-button.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import AppPaths from 'constants/app-paths';
+import { useLocation } from 'react-router-dom';
+import { UseFormReturn } from 'react-hook-form';
+
+type Props = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  form: UseFormReturn<any, any, any>;
+};
+
+export const SubmitButton = ({ form }: Props) => {
+  const location = useLocation();
+  const pathname = location.pathname;
+  const {
+    getValues,
+    formState: { isDirty, isValid },
+  } = form;
+
+  const initButtonDisabled = () => {
+    const formValues = getValues();
+    let isDisabled = true;
+    for (const key in formValues) {
```

Comment:
u can use make an array and use some method for this

Reply:
I added this method for initial disable for the button, but now it works without it. just deleted it

Reply:
it could be a problem with state in the checkbox